x/mongo/driver/batch_cursor.go

```diff
@@ -184,6 +184,15 @@ func NewEmptyBatchCursor() *BatchCursor {
 	return &BatchCursor{currentBatch: new(bsoncore.DocumentSequence)}
 }
 
+// NewBatchCursorFromDocuments returns a batch cursor with current batch set to a sequence-style
+// DocumentSequence containing the provided documents.
+func NewBatchCursorFromDocuments(documents []byte) *BatchCursor {
+	return &BatchCursor{currentBatch: &bsoncore.DocumentSequence{
+		Data:  documents,
+		Style: bsoncore.SequenceStyle,
```

Comment:
Consider adding `id: 0` and/or `server: nil` with a comment that indicates the zero values are important because they indicate the cursor is closed and prevent the Cursor from making any actual `getMore` calls.

Reply:
Good point. Added both and a relevant comment.